rust/worker/src/execution/operators/group.rs

```diff
@@ -0,0 +1,223 @@
+use crate::errors::{ChromaError, ErrorCodes};
+use crate::execution::data::data_chunk::DataChunk;
+use crate::execution::operator::Operator;
+use async_trait::async_trait;
+use std::collections::HashMap;
+use thiserror::Error;
+
+#[derive(Debug)]
+pub struct GroupOperator {
+    // Approximatedly the number of records in a group. Since we are trying to
+    // group the records by id, which can casue the group size to be larger than this
+    // value.
+    pub(crate) group_size: usize,
```

Comment:
Why is this not input but instead a property of the operator? That feels odd to me